### DISCUSSION TITLE
GitHub Check Run placed in the wrong workflow run - UI bug?

### DISCUSSION THREAD

**Discussion by davidelogi** [community/community#39174] [Actions]
### Select Topic Area

Bug

### Body

Hello,

I run into a weird issue.

I have 2 different workflows which are triggered on the same event (actually, one is executing on any push and the second one on push to _main_ branch).

The workflow running on any push is also producing some reports in forms of GitHub Check Run using some marketplace ready-to-use actions (e.g. https://github.com/marketplace/actions/test-reporter).

When this is executing on a push for a dev branch (!= _main_), I can see in the Action _Summary_ nicely on the left pane:
- the name of the job(s) executed
- the list of the generated GH checks

Instead, when this is executing on a push on _main_, in the Action _Summary_ for the same workflow, on the left pane I can see:
- the name of the job(s) executed

But no GH checks!

Investigating, I've found that the GH checks generated now appears **in the other workflow** Action _Summary_, which instead is doing something completely different...

Do you have any explanation for this?

To me this looks like a UI glitch, but please let me know if I may be doing something wrong and I'll try to test on my side.

Thanks!

**Comment by BrianCArnold** [comment]
I'm running into this too, it appears to be a known bug with GitHub Actions that it seems has been ignored for almost 3 years now: 

https://github.com/orgs/community/discussions/24616
